fms/utils/tokenizers.py

```diff
@@ -167,6 +179,184 @@ def encode(self, text, add_special_tokens=False):
+        self.bos_token = "<s>"
+        self.eos_token = "</s>"
+        self.unk_token = "<unk>"
+        self.pad_token = self.eos_token  # Mistral typically uses EOS as pad
+
+    def _load_config(self) -> Dict[str, Any]:
+        """Load configuration from config.json"""
+        config_path = self.model_path / "config.json"
+        if config_path.exists():
+            with open(config_path, "r", encoding="utf-8") as f:
+                return json.load(f)
+        return {}
+
+    def _load_system_prompt(self) -> str:
+        """Load system promp from model path"""
+        system_prompt = self.model_path / "SYSTEM_PROMPT.txt"
+        if system_prompt.exists():
+            with open(system_prompt, "r") as f:
+                return f.read()
+        return ""
+
+    def _load_tokenizer(self) -> Tekkenizer:
+        """
+        Load the tekken tokenizer
+        https://github.com/mistralai/mistral-common/blob/main/src/mistral_common/tokens/tokenizers/tekken.py
+        """
+        # Check for Tekken tokenizer (newer models)
+        tekken_path = self.model_path / "tekken.json"
+        if tekken_path.exists():
+            try:
+                return Tekkenizer.from_file(str(tekken_path))
+            except Exception as e:
+                print(f"Error: Failed to load Tekken tokenizer from {tekken_path}: {e}")
+                raise RuntimeError(
+                    f"Tekkenizer failed to load tekken.json at model_path {self.model_path}"
+                )
+        else:
+            raise RuntimeError(
+                f"Tekkenizer Error: Could not find the tekken.json at model_path {self.model_path}"
+            )
+
+    def encode(self, text: str, add_special_tokens: bool = False) -> List[int]:
+        """Encode a string into a list of token ids.
+
+        Args:
+            text (str): input text string to be tokenized and converted to ids
+            add_special_tokens (bool, optional): prefix bos token to the input text
+
+        Returns:
+            List[int]: token ids of tokenized text string
+        """
+        return self.tokenizer.encode(text, bos=add_special_tokens, eos=False)
+
+    def decode(
+        self,
+        token_ids: List[int],
+        stp: SpecialTokenPolicy = SpecialTokenPolicy.IGNORE,
+    ) -> str:
+        """Decode a list of token ids into a string.
+
+        Args:
+            token_ids (List[int]): The list of token ids to decode.
+            special_token_policy: The policy for handling special tokens.
+                Use the tokenizer's [attribute][mistral_common.tokens.tokenizers.tekken.Tekkenizer.special_token_policy]
+                if `None`. Passing `None` is deprecated and will be changed
+                to `SpecialTokenPolicy.IGNORE` in `mistral_common=1.7.0`.
+
+        Returns:
+            str: Decoded text string
+        """
+        return self.tokenizer.decode(token_ids, stp)
+
+    def tokenize(self, text: str) -> List[str]:
```

Comment:
add that this will be a lot more inefficient than encode for this tokenizer in the deprecation message